### ISSUE TITLE
Type tracking conditionally extended empty tuple

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1915]
This is a tracking bug. The original can be found here: https://github.com/microsoft/pylance-release/issues/1359

**Comment by erictraut** [comment]
A minimal repro:
```python
t: Tuple[int, ...] = tuple()
reveal_type(t) # tuple[_T_co@tuple]
```

Bidirectional type inference logic for constructor needs to have special-case logic for tuple because it is a variadic generic.

**Comment by erictraut** [comment]
This is included in pyright 1.1.145, which I just published. It will also be included in the next release of pylance.
